andela/knights-bn-backend
# #170947667 notification for new request made

#### What does this PR do?
- Notify the manager when a new request is made.
#### Description of Task to be completed?
- for the successful creation of a request, an email is sent to the intended manager
- for the successful creation of a request, an in-app notification is sent to the intended manager
#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git checkout ft-notification-for-new-Request-170947667 ` to switch to this branch.
- run `npm install` to install dependencies.
- go to this link to see live notification `https://codepen.io/king_Bomaye/pen/VwLbQGX?editors=1111`
- create a .env file then add environment variable as shown in ` .env.example`,  check our slack channel `code-knights-12` for .env

```
   MANAGER_PSWD=$2b$10$SO7hwwylEnHKZIv68w/ePOK4yybPnhXa5.tQSUBe.PSb6FBKyuVO.
```

- create two Postgres database for development and then test
- run `npm run createTables` to create tables for you

- run `npm run dev-start` to start the server.

- login using admin credentials POST `http://localhost:4000/api/v1/auth/login`

`{
    "email": "william.ishimwe@andela.com",
    "password": "Password@1"
}`

- copy the `manager's token` in the following app `https://codepen.io/king_Bomaye/pen/VwLbQGX?editors=1111`

<img width="747" alt="Screen Shot 2020-02-28 at 20 48 29" src="https://user-images.githubusercontent.com/48021034/75578035-cb7a1e80-5a6b-11ea-9837-e2bc4dacb95f.png">


- signup a regular user user by POST `localhost:4000/api/v1/auth/signup`
{
                 "firstName": "Mugenzi",
                  "lastName": "Jean",
                  "gender": "male",
                  "passportNumber": "ui8908dv",
                 "email": "david@andela.com",
                 "password": "Niyonkuru@1"
            }

- make a new request by POST `http://localhost:4000/api/v1/trips/returnTrip`

{
	"origin": "Burundo",
	"destination": "kgl",
	"departureDate": "9002-04-02",
	"returnDate": "9003-05-18",
	"accommodation": "XYZ campus",
	"reason": "Having a meeting",
	"passportNumber": "yuyut56"
}

- will see a successful message for success request and an email will be sent to the manager 


#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

- [#170947667](https://www.pivotaltracker.com/story/show/170947667)

#### Screenshots (if appropriate)
              

Successfully created notification(I have to create a small UI to demonstrate successful notification)

<img width="1238" alt="Screen Shot 2020-02-27 at 16 12 27" src="https://user-images.githubusercontent.com/48021034/75453157-23395c80-597c-11ea-9de1-4eb204100827.png">


 Email notification send to the Manager(William in this case is the manager)

<img width="757" alt="Screen Shot 2020-02-27 at 11 24 07" src="https://user-images.githubusercontent.com/48021034/75438344-73a2c100-5960-11ea-9af9-baa06f2cbfe8.png">


#### Questions:
N/A
